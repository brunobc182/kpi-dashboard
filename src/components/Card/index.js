import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { COLORS, DEVICE } from "../../theme";

/**
 * This styled components is responsible for our main
 * card style. The property background allow us to
 * make the two circles working according of our
 * design specs.
 */

const Wrapper = styled.div`
  max-width: 300px;
  width: 100%;
  background: radial-gradient(
      circle at 100% 50px,
      rgba(255, 255, 255, 0.3) 0%,
      rgba(255, 255, 255, 0.3) 20%,
      transparent 10%
    ),
    radial-gradient(
      circle at 250px 150px,
      rgba(255, 255, 255, 0.3) 0%,
      rgba(255, 255, 255, 0.3) 30%,
      transparent 10%
    ),
    ${({ bgColor }) => bgColor};
  padding: 40px;
  border-radius: 4px;
  margin: 16px;

  @media ${DEVICE.TABLET} {
    &:last-child {
      margin-right: 0;
    }
  }
`;

const Title = styled.h4`
  color: ${COLORS.WHITE};
  margin: 0;
  font-weight: 500;
`;

const CurrentPeriod = styled.p`
  color: ${COLORS.WHITE};
  margin: 16px 0;
  font-size: 32px;
  font-weight: 700;
`;

const LastPeriod = styled.span`
  color: ${COLORS.WHITE};
  margin: 0;
`;

/**
 * Represents the component Card.
 *
 * @param {string} title - The title of the card.
 * @param {number} currentPeriod - The current period of the card.
 * @param {number} lastPeriod - The lastPeriod of the card.
 * @param {string} bgColor - Thhe main color of card background.
 */
function Card({ title, currentPeriod, lastPeriod, bgColor, ...props }) {
  return (
    <Wrapper bgColor={bgColor}>
      <Title>{title}</Title>
      <CurrentPeriod>{currentPeriod}</CurrentPeriod>
      <LastPeriod>{`${lastPeriod} Last period`}</LastPeriod>
    </Wrapper>
  );
}

Card.propTypes = {
  title: PropTypes.string,
  currentPeriod: PropTypes.number,
  lastPeriod: PropTypes.number,
  bgColor: PropTypes.string,
};

Card.defaultProps = {
  title: "Card title",
  currentPeriod: 20,
  lastPeriod: 30,
  bgColor: `${COLORS.COD_GRAY}`,
};

export default Card;
