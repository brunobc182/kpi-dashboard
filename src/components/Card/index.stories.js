import React from "react";
import Card from "./";

export const Default = () => <Card />;
export const WithProps = () => (
  <Card
    bgColor="#e62a19"
    title="Active sourcing"
    currentPeriod={10}
    lastPeriod={15}
  />
);

export default {
  title: "Components/Card",
  component: Card,
};
