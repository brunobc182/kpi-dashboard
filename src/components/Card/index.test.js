import React from "react";
import { render } from "@testing-library/react";
import Card from "./";

describe("Card", () => {
  it("renders correctly", () => {
    const { container } = render(<Card />);

    expect(container).toMatchSnapshot();
  });

  it("renders correctly with default values", () => {
    const { container, getByText } = render(<Card />);
    const title = getByText(/Card title/i);
    const currentPeriod = getByText(/20/i);
    const lastPeriod = getByText(/30/i);

    expect(title).toBeInTheDocument();
    expect(currentPeriod).toBeInTheDocument();
    expect(lastPeriod).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders correctly with given props", () => {
    const { container, getByText } = render(
      <Card
        bgColor="#e62a19"
        title="Active sourcing"
        currentPeriod={10}
        lastPeriod={15}
      />
    );
    const title = getByText(/Active sourcing/i);
    const currentPeriod = getByText(/10/i);
    const lastPeriod = getByText(/15/i);

    expect(title).toBeInTheDocument();
    expect(currentPeriod).toBeInTheDocument();
    expect(lastPeriod).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
