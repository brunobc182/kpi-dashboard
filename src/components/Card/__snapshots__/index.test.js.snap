// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div>
  <div
    class="sc-AxjAm hhglcc"
  >
    <h4
      class="sc-AxirZ ddxAos"
    >
      Card title
    </h4>
    <p
      class="sc-AxiKw fhzMTD"
    >
      20
    </p>
    <span
      class="sc-AxhCb kbojYk"
    >
      30 Last period
    </span>
  </div>
</div>
`;

exports[`Card renders correctly with default values 1`] = `
<div>
  <div
    class="sc-AxjAm hhglcc"
  >
    <h4
      class="sc-AxirZ ddxAos"
    >
      Card title
    </h4>
    <p
      class="sc-AxiKw fhzMTD"
    >
      20
    </p>
    <span
      class="sc-AxhCb kbojYk"
    >
      30 Last period
    </span>
  </div>
</div>
`;

exports[`Card renders correctly with given props 1`] = `
<div>
  <div
    class="sc-AxjAm dxEya-d"
  >
    <h4
      class="sc-AxirZ ddxAos"
    >
      Active sourcing
    </h4>
    <p
      class="sc-AxiKw fhzMTD"
    >
      10
    </p>
    <span
      class="sc-AxhCb kbojYk"
    >
      15 Last period
    </span>
  </div>
</div>
`;
